framework/src/modules/validators/api.ts

```diff
@@ -39,21 +52,54 @@ export class ValidatorsAPI extends BaseAPI {
 		generatorKey: Buffer,
 		proofOfPossession: Buffer,
 	): Promise<boolean> {
-		const validatorsSubStore = apiContext.getStore(this.moduleID, STORE_PREFIX_VALIDATORS_DATA);
-
+		const validatorsSubStore = apiContext.getStore(this.moduleID, SUBSTORE_PREFIX_VALIDATORS_DATA);
 		const addressExists = await validatorsSubStore.has(validatorAddress);
 		if (addressExists) {
-			return false;
+			apiContext.eventQueue.add(
+				this.moduleName,
+				EVENT_NAME_GENERATOR_KEY_REGISTRATION,
+				codec.encode(generatorKeyRegDataSchema, {
+					generatorKey,
+					result: KEY_REG_RESULT_ALREADY_VALIDATOR,
+				}),
+				[validatorAddress],
+				true,
```

Comment:
not sure if its non-revertible, also at other places, @shuse2 ?

Reply:
in the LIP, `non-revertible` is mentioned as `emitPersistentEvent`. so when it says `emitPersistentEvent` it should be `true` but otherwise, it should be `false`

Reply:
then it looks good 👍